pkg/backends/calico/routes.go

```diff
@@ -46,15 +46,32 @@ type routeGenerator struct {
 	svcInformer, epInformer cache.Controller
 	svcIndexer, epIndexer   cache.Indexer
 	svcRouteMap             map[string][]string // maps service name to ip
+	clusterCIDR             string
 }
 
 // NewRouteGenerator initializes a kube-api client and the informers
-func NewRouteGenerator(c *client) (rg *routeGenerator, err error) {
+func NewRouteGenerator(c *client, clusterCIDR string) (rg *routeGenerator, err error) {
+	// Parse clusterCIDr to make sure it is valid.
```

Comment:
Moved the clusterCIDR parsing here rather than Start().

We wanted the `clusterCIDR` as a field on the RG and it made sense to do that in the initializer.